packages/venia-concept/upward.yml

```diff
@@ -33,11 +33,10 @@ response:
     - matches: request.url.pathname
       pattern: '^/(graphql|rest)(/|$)'
       use: proxy
-    # Requests to create an account are handled by the top-level 'index'
-    # object, which returns a generic app shell
+    # Requests to create an account are handled by the 'appShell' object.
     - matches: request.url.pathname
```

Comment:
Also, it's not clear to me why lines 37-39 are required. I would have assumed we could just resolve similar to how we resolve `/search.html`, but if I remove these lines I get failures like `unable to load "<url>/dist/create-account"`

Reply:
Actually, you've uncovered an optimization opportunity in the `resource` definition. Matches are evaluated top to bottom, so we want the hardcoded matches to be _above_ the matches that check `urlResolver`. When we know ahead of time that the hardcoded URL will not be resolved with urlResolver, we want to avoid evaluating urlResolver, since it's actually placing a graphQL query. There's gotta be a simpler way to do that in the future.

For now, to prevent merge conflicts, let's just move all the match clauses against literal URL paths to the _top_ of the conditional stack, and all the matches against `urlResolver` under them. That helps us avoid an additional request.

Reply:
Ok, moved.